### PR TITLE
bug 1548385 using .gitignore with a secured git repo

### PR DESCRIPTION
**Pull request by kalexand-rh**
https://bugzilla.redhat.com/show_bug.cgi?id=1548385

### CODE REVIEW

**Comment by kalexand-rh** [comment]
@wzheng1, PTAL

File: dev_guide/builds/build_inputs.adoc [hunk 1]
```diff
@@ -458,6 +458,52 @@ section in your *_.gitconfig_* file:
...
+--from-file=client.key=/var/run/secrets/openshift.io/source/client.key
+----
+<1> The user's Git user name.
+<2> The password for this user.
+
+[IMPORTANT]
+====
+To avoid having to enter your password, be sure to specify the S2I image in 
+your builds.
+====
```
Comment (wzheng1) [inline dev_guide/builds/build_inputs.adoc]: Suggest to make it more clear about this part: If your local git configuration is NOT able to git clone the repository, specifying the S2I image will avoid to promote asking you to provide username and password.

File: dev_guide/builds/build_inputs.adoc [hunk 2]
```diff
@@ -458,6 +458,53 @@ section in your *_.gitconfig_* file:
...
+--from-file=.gitconfig=.gitconfig \
+--from-file=client.crt=/var/run/secrets/openshift.io/source/client.crt \
+--from-file=cacert.crt=/var/run/secrets/openshift.io/source/cacert.crt \
+--from-file=client.key=/var/run/secrets/openshift.io/source/client.key
+----
+<1> The user's Git user name.
+<2> The password for this user.
+
+[IMPORTANT]
+====
```
Comment (kalexand-rh) [inline dev_guide/builds/build_inputs.adoc]: @wzheng1, does this sound better?

Comment (wzheng1) [reply dev_guide/builds/build_inputs.adoc]: Yes, thanks! acked by wzheng.

Comment (kalexand-rh) [reply dev_guide/builds/build_inputs.adoc]: Great! Thank you. :) Merging.

**Comment by kalexand-rh** [comment]
@openshift/team-documentation, PTAL

File: dev_guide/builds/build_inputs.adoc [hunk 3]
```diff
@@ -458,6 +458,53 @@ section in your *_.gitconfig_* file:
 ----
 ====
 
+[[source-secrets-gitconfig-file-secured]]
+==== .gitconfig File for Secured Git
+
+If your Git server is secured with 2-way SSL and user name with password
+you must add the certificate files to your source build and add references to
+them in the *_.gitconfig_* file:
+
```
Comment (mburke5678) [inline dev_guide/builds/build_inputs.adoc]: s/them/the certificate files ??

File: dev_guide/builds/build_inputs.adoc [hunk 4]
```diff
@@ -458,6 +458,53 @@ section in your *_.gitconfig_* file:
...
+. Add the *_client.crt_*, *_cacert.crt_*, and *_client.key_* files to the 
+*_/var/run/secrets/openshift.io/source/_* folder in the
+xref:../../dev_guide/application_lifecycle/new_app.adoc#specifying-source-code[application
+source code].
+
+. In the *_.gitconfig_* file for the application, add the `[http]` section
+shown in the following example:
++
```
Comment (mburke5678) [inline dev_guide/builds/build_inputs.adoc]: @kalexand-rh My ignorance, perhaps. Is the .gitconfig file for the server, repo. application?
Above you talk about the server and gitconfig; here the git config _for the application_.

**Comment by mburke5678** [comment]
@kalexand-rh One nit; one question; otherwise LGTM

File: dev_guide/builds/build_inputs.adoc (hunk 4 above)
Comment (kalexand-rh) [reply dev_guide/builds/build_inputs.adoc]: The git config file is in the repo that holds your application code. The git server is the instance that holds all the git repos. If I'm reading it right, the certificate files are good for the whole server, but you need to put the references to them in each application repo's git config.

Comment (mburke5678) [reply dev_guide/builds/build_inputs.adoc]: So it is the gitconfig for the repo where the application is held/resides/is stored. That is what I thought. 
What you have is probably fine.

Comment (kalexand-rh) [reply dev_guide/builds/build_inputs.adoc]: @wzheng1, will you please confirm that the gitconfig file is for the application, and that the gitconfig file is not for the entire Git server?

Comment (wzheng1) [reply dev_guide/builds/build_inputs.adoc]: @kalexand-rh  the gitconfig file is for the entire Git server.

**Comment by kalexand-rh** [comment]
I've fixed the nit and addressed the question. Thank you. :)

**Pull request merged**